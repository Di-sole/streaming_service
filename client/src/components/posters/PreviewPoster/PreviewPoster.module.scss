.container {
    display: block;
    width: 153.14px;

    //max-width: 200px;

    &__block {
        border-radius: 8px;
        overflow: hidden;

        &_image {
            position: relative;
            margin-bottom: 8px;
            background-color: #e6e6e6;
            transition: 0.3s;

            //height: 235.05px;

            &:hover {
                transform-origin: bottom;
                transform: scale(103%, 103%);
            }
        }
    }

    &__imageWrapper {
        width: 100%;
        padding-top: 165%;
        position: relative;
        border-radius: 50%;
    }

    &__image {
        position: absolute;
        top: 0;
        left: 0;
        bottom: 0;
        right: 0;
        /* object-fit: cover;*/
    }

    &__title {
        font-size: 15px;
        font-weight: 600;
        line-height: 120%;
        color: #d9d7e0;
        overflow: hidden;
        white-space: nowrap;
        text-overflow: ellipsis;
    }

    &__badge {
        position: absolute;
        bottom: 4px;
        right: 4px;
    }
}
