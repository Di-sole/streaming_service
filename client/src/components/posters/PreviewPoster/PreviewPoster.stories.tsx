import { Meta, StoryObj } from '@storybook/react';
import './PreviewPoster.module.scss';
import Poster from './PreviewPoster';
import IMovies from '@/models/IMovies';

const meta: Meta<typeof Poster> = {
    title: 'Poster/PreviewPoster',
    component: Poster,
};

export default meta;

type Story = StoryObj<typeof Poster>;

const data: IMovies = {
    id: 1,
    name: 'Название',
    enName: 'Название',
    posterPreviewURL: require('../../../testAsserts/img/BG554460.jpg'),
    premiereRussia: '',
    hasIMAX: false,
    year: 1900,
    ageRating: 0,
    ratingKp: 9.5,
    votesKp: 11111,
    movieLength: 126,
    genres: [
        { id: 1, name: 'Жанр', enName: 'Жанр' },
        { id: 2, name: 'Жанр', enName: 'Жанр' },
    ],
    countries: [{ id: 1, name: 'Страна', enName: 'Страна' }],
};

export const PreviewPoster: Story = {
    args: {
        movie: data,
    },
};
