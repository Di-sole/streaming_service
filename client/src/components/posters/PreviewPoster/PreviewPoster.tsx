import AgeBadge from '@/components/UI/badges/AgeBadge/AgeBadge';
import PriceBadge from '@/components/UI/badges/PriceBadge/PriceBadge';
import PreviewPosterContent from '../PreviewPosterContent/PreviewPosterContent';
import IMovies from '@/models/IMovies';
import Image from 'next/image';
import Link from 'next/link';
import style from './PreviewPoster.module.scss';
import { useRouter } from 'next/router';

interface PreviewPosterProps {
    movie: IMovies;
    className: string;
}

const PreviewPoster = ({ movie, className }: PreviewPosterProps) => {
    const { locale } = useRouter();
    return (
        <Link className={[style.container, className].join(' ')} href="/">
            <div className={[style.container__block, style.container__block_image].join(' ')}>
                <div className={style.container__imageWrapper}>
                    <Image className={style.container__image} src={movie.posterpreviewUrl} alt={movie.name} fill />
                </div>
                <PreviewPosterContent movie={movie} />
                <AgeBadge className={style.container__badge} value={`${movie.ageRating}`} />
            </div>

            <div className={[style.container__block, style.container__block_text].join(' ')}>
                <p className={style.container__title}>{locale === 'en' && movie.enName ? movie.enName : movie.name}</p>
                <PriceBadge priceType="purchase" />
            </div>
        </Link>
    );
};

export default PreviewPoster;
