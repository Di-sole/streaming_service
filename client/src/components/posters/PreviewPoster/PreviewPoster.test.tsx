import { render, screen } from '@testing-library/react';
import PreviewPoster from './PreviewPoster';
import { moviesTest } from '../../../testAsserts/testItems';

jest.mock('next-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

jest.mock('next/router', () => ({
    useRouter() {
        return {
            pathname: '',
        };
    },
}));

describe('PREVIEWPOSTER SNAPSHOTS TESTS', () => {
    test('PreviewPoster should not be changed', () => {
        render(<PreviewPoster movie={moviesTest[0]} className="previewItem" />);

        const link = screen.getByRole('link');
        expect(link).toMatchSnapshot();
    });
});
