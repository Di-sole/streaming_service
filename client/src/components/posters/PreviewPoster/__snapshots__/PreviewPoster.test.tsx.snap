// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PREVIEWPOSTER SNAPSHOTS TESTS PreviewPoster should not be changed 1`] = `
<a
  class="container previewItem"
  href="/movies/535341"
>
  <div
    class="container__block container__block_image"
  >
    <div
      class="container__imageWrapper"
    >
      <img
        alt="1+1"
        class="container__image"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_535341.jpg&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_535341.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_535341.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_535341.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_535341.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_535341.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_535341.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_535341.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_535341.jpg&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
    <div
      class="container"
    >
      <div
        class="container__icons"
      >
        <img
          alt="icon"
          class="container__icon container__icon_favorite"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="icon"
          class="container__icon container__icon_similar"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="icon"
          class="container__icon container__icon_rate"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="icon"
          class="container__icon container__icon_dislike"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </div>
      <div
        class="container container__info"
      >
        <div
          class="container__rating"
        >
          <span
            class="container__rating container__rating_integer"
          >
            8
          </span>
          <span
            class="container__rating container__rating_fraction"
          >
            .8
          </span>
        </div>
        <div
          class="container__text"
        >
          <p>
            2011
            , 
            France
            ,
             
            Drama
          </p>
          <p>
            112 minutes.2
          </p>
        </div>
      </div>
    </div>
    <div
      class="badge container__badge"
    >
      16
      +
    </div>
  </div>
  <div
    class="container__block container__block_text"
  >
    <p
      class="container__title"
    >
      1+1
    </p>
  </div>
</a>
`;
